src/device/device_io_hid.cpp

```diff
@@ -44,7 +44,8 @@ namespace hw {
     
     static std::string safe_hid_error(hid_device *hwdev) {
       if (hwdev) {
-        return  std::string((char*)hid_error(hwdev));
+        const char* error_str = (const char*)hid_error(hwdev);
+        return  std::string(error_str == nullptr ? "Unknown error" : error_str);
```

Comment:
The API seems to return wchar_t*, so the string conversion won't work (before your patch).
The patch can still go in though.

Reply:
I noticed it returned `wchar_t*`, however, from what I know and my own experiments with Qt and its `QString`, which uses wide characters underneath, casting to an 8-bit string should be OK. Windows and Linux, AFAIK, both will interpret the characters correctly. If there's a better way to do this conversion, let's do that.

Reply:
AFAIK wchar_t is UTF-16 (or maybe UCS-4). But maybe the lib is returning UTF-8 in there, which would be surprising, but possible.

Reply:
> AFAIK wchar_t is UTF-16

Right, and the lib returns UTF-16 as well. So the conversion won't work. But as you already mentioned this conversion "bug" was there before the current patch.